testssl.sh

```diff
@@ -6283,9 +6283,16 @@ run_server_preference() {
           pr_svrty_high "nope (NOT ok)"; out " -- only for TLS 1.3"
           fileout "$jsonID" "HIGH" "server -- < TLS 1.3 client determined"
      else
-          pr_svrty_best "yes (OK)"
-          "$has_tls13_cipher_order" && out " -- TLS 1.3 and below"
-          fileout "$jsonID" "OK" "server"
+          if "$has_tls13_cipher_order"; then
+               if "$TLS13_ONLY"; then
+                    out "yes (TLS 1.3 only)"
+                    fileout "$jsonID" "INFO" "server (TLS 1.3)"
+               else
+                    pr_svrty_best "yes (OK)"
+                    out " -- TLS 1.3 and below"
+                    fileout "$jsonID" "OK" "server"
+               fi
+          fi
      fi
```

Comment:
I believe that there is a problem here. Given the above code, if a server enforces a cipher order but does not support TLS 1.3, nothing is printed.

The code above applies whenever the server enforces a cipher order for all supported protocol versions. Given the previous `if` statements, `$has_cipher_order` must be `true` and either (1) the server supports TLS 1.3 and `$has_tls_cipher_order` is `true` or (2) the server does not support TLS 1.3 and `$has_tls_cipher_order` is `false`.

The above code handles the case in which the server supports TLS 1.3, but not the case in which TLS 1.3 isn't supported.